zipline/utils/calendars/trading_schedule.py

```diff
@@ -0,0 +1,379 @@
+#
+# Copyright 2016 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import (
+    ABCMeta,
+    abstractmethod,
+    abstractproperty,
+)
+from functools import partial
+
+from zipline.utils.memoize import remember_last
+
+from .exchange_calendar import get_calendar
+from .calendar_helpers import (
+    next_scheduled_day,
+    previous_scheduled_day,
+    next_open_and_close,
+    previous_open_and_close,
+    scheduled_day_distance,
+    minutes_for_day,
+    days_in_range,
+    minutes_for_days_in_range,
+    add_scheduled_days,
+    all_scheduled_minutes,
+    next_scheduled_minute,
+    previous_scheduled_minute,
+)
+
+
+class TradingSchedule(object):
+    """
+    A TradingSchedule defines the execution timing of a TradingAlgorithm.
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self):
+        # Assign the partial calendar helpers
+        self.next_execution_day = partial(
```

Comment:
What does that have to do with implementing these as partials?  I'm asking why the implementation of this isn't something like:

```
def next_execution_day(self, date):
    return next_scheduled_day(date, last_trading_day=self.last_execution_day, is_scheduled_day_hook=self.is_executing_on_day)
```